docs/3-MODELS.md

```diff
@@ -0,0 +1,124 @@
+## 2. MODELS (Storing the collected data)
+
+The data collected from various sources, as discussed in [2-SOURCES](/docs/2-SOURCES.md), is stored in a PostgreSQL
+database. The tables are designed using Django models.
+
+All of the project's models are defined inside the app: `ports`. The models are split into three different files: `port`,
+`buildhistory` and `stats`. Each file contains the models related to the category defined by file name.
+
+#### 1. `Port` Model
+
+It stored the basic information related to a port, with the following fields:
+- `portdir`
+- `description`
+- `homepage`
+- `epoch`
+- `platforms`
+- `categories` - A many-to-many relation with `Category` model
+- `long_description`
+- `version`
+- `revision`
+- `closedmaintainer`
+- `name`
+- `license`
+- `replaced_by` - Default values is `Null` for active ports. If the value is not null, the port is considered as obsolete.
+- `active` - Value is `False` for deleted ports, otherwise `True`*.
+
+***A Port is marked as deleted when the port is available in the app's database, but does not exist in PortIndex. If a 
+port with the same name is added later in time, then previous object is replaced with new object and the newer ports gets
+marked as `active=True` automatically.*
+
+#### 2. `Category` Model
+
+Contains the categories in which the ports is divided. It is related with the `Port` model using a many-to-many relation.
+The model has a single field `name` which also becomes the primary key for the table.
+
+#### 3. `Maintainer` Model
+
+Stores email and github-handle of the maintainers, and maps them with objects from the `Port` model.
+
+- `name`
+- `domain`
+- `github`
+- `ports`- many-to-many relation with the `Port` Model.
+
+*The `name`, `domain`, `github` triplet is unique. But due to inconsistencies in the `Portfiles`, there might be cases
+when we have two entries with same github-handle but different emails. In such cases, a warning is displayed on the
+maintainer's page.*
```

Comment:
I'm not sure if I ever saw such a warning. But what would be more useful would be a page displaying all such inconsistencies, so that we can easily fix them, else it's pretty impossible to find one. Not super high priority, but something to note down.

Reply:
Here is an example: https://ports.macports.org/maintainer/github/yan12125/